enhancements/cluster-logging/loki-gateway-configuration.md

```diff
@@ -0,0 +1,517 @@
+      oidc:
+        issuerURL: https://127.0.0.1:5556/dex
+        redirectURL: https://localhost:8443/oidc/tenant-a/callback
+        usernameClaim: test
+        groupClaim: test
+    authorization:
+      rbac:
+        roleBindings:
+        - name: tenant-a
+          roles:
+          - read-write
+          subjects:
+          - kind: user
+            name: admin@example.com
+        roles:
+        - name: read-write
+          permissions:
+          - read
+          - write
+          resources:
+          - metrics
+          tenants:
+          - tenant-a
+```
+
+The LokiStack Gateway deployment will need to create a static rego file which will include:
+
+```rego
+package observatorium
+
+import input
+import data.roles
+import data.roleBindings
+
+default allow = false
+
+allow {
+  some roleNames
+  roleNames = roleBindings[matched_role_binding[_]].roles
+  roles[i].name == roleNames[_]
+  roles[i].resources[_] = input.resource
+  roles[i].permissions[_] = input.permission
+  roles[i].tenants[_] = input.tenant
+}
+
+matched_role_binding[i] {
+  roleBindings[i].subjects[_] == {"name": input.subject, "kind": "user"}
+}
+
+matched_role_binding[i] {
+  roleBindings[i].subjects[_] == {"name": input.groups[_], "kind": "group"}
+}
+```
+
+### LokiStack CR configuration - _Dynamic Mode_
+
+In this mode, user will need to configure tenants, oidc and an OPA URL madatorily in the LokiStack CR. This OPA url will delegate authorization of OIDC extracted subject information to a third-party OPA-compatible endpoint.
+
+During the reconcilation of LokiStack CR, the user provided details will be validated. If anything is missing or provided extra, the Loki operator will give out an error.
+
+The LokiStack CR will look like this:
+
+```yaml
+tenants:
+    mode: dynamic
+    authentication:
+    - name: tenant-a
+      oidc:
+        issuerURL: https://127.0.0.1:5556/dex
+        redirectURL: https://localhost:8443/oidc/tenant-a/callback
+        usernameClaim: test
+        groupClaim: test
+    authorization:
+      opa:
+        url: http://127.0.0.1:8080/v1/data/observatorium/allow
+```
+
+### LokiStack CR configuration - _Openshift Logging Mode_
+
+In this mode, user will need to configure nothing as everything will be done by Loki operator itself.
+
+During the reconcilation of LokiStack CR, if user provides anything other than `mode`, the Loki operator will give out an error.
+
+The LokiStack CR which will look like this:
+
+```yaml
+tenants:
+    mode: openshift-logging
+```
+
+The LokiStack Gateway deployment will need to configure everything for the user:
+
+```yaml
+tenants:
+- name: application
+  id: 32e45e3e-b760-43a2-a7e1-02c5631e56e9
+  oidc:
+    clientID: test
+    clientSecret: test
+    issuerCAPath: ./tmp/certs/ca.pem
+    issuerURL: https://127.0.0.1:5556/dex
+    redirectURL: https://localhost:8443/oidc/application/callback
+    usernameClaim: name
+  opa:
+    url: http://127.0.0.1:8080/v1/data/observatorium/allow
```

Comment:
Let's remove the references to observatorium here as we did in the implementation